onedal/tests/test_common.py

```diff
@@ -15,26 +15,46 @@
 # ==============================================================================
 
 import os
+import pkgutil
 from glob import glob
 
-import pytest
 
-
-def test_sklearn_check_version_ban():
-    """This test blocks the use of sklearn_check_version
-    in onedal files. The versioning should occur in the
-    sklearnex package for clarity and maintainability.
+def _check_primitive_usage_ban(primitive_name, package, allowed_locations=None):
+    """This test blocks the usage of the primitive in
+    in certain files.
     """
-    from onedal import __file__ as loc
+
+    # TODO:
+    # Address deprecation warning.
+    # The function "get_loader" is deprecated Use importlib.util.find_spec() instead.
+    # Will be removed in Python 3.14.
+    loc = pkgutil.get_loader(package).get_filename()
```

Comment:
```suggestion
    loc = importlib.util.find_spec(package).origin
```
I'm sorry to have made the previous suggestion. This should solve the todo, and thank you very much for putting in the effort on the changes.